src/dune_rules/dialect.ml

```diff
@@ -77,10 +77,14 @@ let decode =
         (map
            ~f:(fun (loc, x) -> (loc, x, []))
            (located Action.decode_dune_file))
-    in
+    and+ syntax_ver = Syntax.get_exn Stanza.syntax in
     let extension =
-      if String.contains extension '.' then
-        User_error.raise ~loc [ Pp.textf "extension must not contain '.'" ];
+      if String.length extension > 0 && extension.[0] = '.' then
```

Comment:
Would it be possible to share this with the extension parsing code in `melange_stanzas`?  Some consistency around extensions would be very good to have. I would also suggest for such generic extension code to be moved to `dune_lang`.

Reply:
Yes, I pushed a commit sharing this code with `Melange_stanzas`, as suggested. Thanks!